### PR TITLE
MDEV-27861: Creating partitioned tables should not be allowed with ws…

### PR DESCRIPTION
**Pull request by janlindstrom**
…rep_osu_method=TOI and wsrep_strict_ddl=ON

Problem was incorrect handling of partitioned tables. If partition handler is InnoDB all DDL should be allowed even with wsrep_strict_ddl. For other storage engines DDL should not be allowed and error should be issued.

### CODE REVIEW

File: sql/sql_table.cc [hunk 1]
```diff
@@ -10155,9 +10167,20 @@ bool mysql_alter_table(THD *thd, const LEX_CSTRING *new_db,
   if (WSREP(thd) &&
       (thd->lex->sql_command == SQLCOM_ALTER_TABLE ||
        thd->lex->sql_command == SQLCOM_CREATE_INDEX ||
-       thd->lex->sql_command == SQLCOM_DROP_INDEX) &&
-      !wsrep_should_replicate_ddl(thd, table_list->table->s->db_type()->db_type))
-    DBUG_RETURN(true);
+       thd->lex->sql_command == SQLCOM_DROP_INDEX))
+  {
+    /*
+       For partitioned tables we need to resolve implenting
```
Comment (sciascid) [inline sql/sql_table.cc:10173]: Typo: implenting -> implementing
And a suggestion: better word would be "underlying" storage engine?

File: sql/sql_table.cc [hunk 2]
```diff
@@ -2471,11 +2471,23 @@ int mysql_rm_table_no_locks(THD *thd, TABLE_LIST *tables, bool if_exists,
...
     {
-      if (WSREP(thd) && hton && !wsrep_should_replicate_ddl(thd, hton->db_type))
+#ifdef WITH_WSREP
+      if (WSREP(thd) && hton)
       {
-        error= 1;
-        goto err;
-      }
+         /*
+           For partitioned tables we need to resolve implenting
```
Comment (sciascid) [inline sql/sql_table.cc:2478]: Same typo as above

File: sql/sql_table.cc [hunk 3]
```diff
@@ -2471,11 +2471,23 @@ int mysql_rm_table_no_locks(THD *thd, TABLE_LIST *tables, bool if_exists,
...
+           storage engine.
+         */
+         const handlerton *ht = table->table ?
+           table->table->file->partition_ht() ?
+             table->table->file->partition_ht() : hton : hton;
```
[the remark this answers is not in the document]
Comment (janlindstrom) [reply sql/sql_table.cc]: Fixed

File: sql/sql_table.cc (hunk 1 above)
Comment (janlindstrom) [reply sql/sql_table.cc]: Fixed

File: sql/sql_table.cc (hunk 2 above)
Comment (janlindstrom) [reply sql/sql_table.cc]: Fixed

**Pull request closed without merging**